cf-agent/package_module.c

```
@@ -0,0 +1,1611 @@
+            if (StringSafeEqual(type, "file"))
+            {
+                package_data->type = PACKAGE_TYPE_FILE;
+            }
+            else if (StringSafeEqual(type, "repo"))
+            {
+                package_data->type = PACKAGE_TYPE_REPO;
+            }
+            else
+            {
+                Log(LOG_LEVEL_VERBOSE, "unsupported package type: %s", type);
+                free(package_data);
+                return NULL;
+            }
+        }
+        else if (StringStartsWith(line, "Name="))
+        {
+            if (package_data->name)
+            {
+                /* Some error occurred as we already have name for 
+                 * given package. */
+                Log(LOG_LEVEL_ERR, "Duplicated package name "
+                    "received: [%s] %s", line, package_data->name);
+                free(package_data);
+                return NULL;
+            }
+            package_data->name = 
+                SafeStringDuplicate(line + strlen("Name="));
+        }
+        else if (StringStartsWith(line, "Version="))
+        {
+            if (package_data->version)
+            {
+                /* Some error occurred as we already have version for 
+                 * given package. */
+                Log(LOG_LEVEL_ERR, "Duplicated package version "
+                    "received: [%s] %s", line, package_data->version);
+                free(package_data);
+                return NULL;
+            }
+            package_data->version = 
+                SafeStringDuplicate(line + strlen("Version="));
+        }
+        else if (StringStartsWith(line, "Architecture="))
+        {
+            if (package_data->arch)
+            {
+                /* Some error occurred as we already have arch for 
+                 * given package. */
+               Log(LOG_LEVEL_ERR, "Duplicated package architecture "
+                    "received: [%s] %s", line, package_data->arch);
```

Given that we haven't checked the new arch is the same as the previously given one (and, if we did that check, there'd be a case for ignoring the error), the log message mis-describes the problem; the architecture isn't **Duplicated**.  A second package architecture has been received and we don't like that.  `s/Duplicated/Extraneous/g` would perhaps be more apt.  Applies to many other Log() messages in this function.